client/hosting/server-settings/components/sftp-card/index.tsx

```diff
@@ -39,53 +39,96 @@ import { getSelectedSiteId, getSelectedSiteSlug } from 'calypso/state/ui/selecto
+	withAnalytics(
+		composeAnalytics(
+			recordGoogleEvent( 'Hosting Configuration', 'Clicked "Reset Password" Button in SFTP Card' ),
+			recordTracksEvent( 'calypso_hosting_configuration_reset_sftp_password' ),
+			bumpStat( 'hosting-config', 'reset-sftp-password' )
+		),
+		resetAtomicSftpPassword( siteId, sshUsername )
+	);
 
-const SftpQuestionsContainer = styled.div( {
-	marginBottom: '1.5em',
-} );
+const createSftpUser = ( siteId: number | null, currentUserId: number | null ) =>
+	withAnalytics(
+		composeAnalytics(
+			recordGoogleEvent(
+				'Hosting Configuration',
+				'Clicked "Create SFTP Credentials" Button in SFTP Card'
+			),
+			recordTracksEvent( 'calypso_hosting_configuration_create_sftp_user' ),
+			bumpStat( 'hosting-config', 'create-sftp-user' )
+		),
+		createAtomicSftpUser( siteId, currentUserId )
+	);
 
-const SftpPasswordExplainer = styled.p( {
-	marginBottom: '8px',
-} );
+const enableSshAccess = ( siteId: number | null ) =>
+	withAnalytics(
+		composeAnalytics(
+			recordTracksEvent( 'calypso_hosting_configuration_enable_ssh_access' ),
+			bumpStat( 'hosting-config', 'enable-ssh-access' )
+		),
+		enableAtomicSshAccess( siteId )
+	);
 
-const SftpEnableWarning = styled.p( {
-	color: 'var(--color-text-subtle)',
-} );
+const disableSshAccess = ( siteId: number | null ) =>
+	withAnalytics(
+		composeAnalytics(
+			recordTracksEvent( 'calypso_hosting_configuration_disable_ssh_access' ),
+			bumpStat( 'hosting-config', 'disable-ssh-access' )
+		),
+		disableAtomicSshAccess( siteId )
+	);
+
+type SftpCardProps = {
+	disabled: boolean;
+};
 
-const SftpSshLabel = styled( FormLabel )( {
-	marginTop: '16px',
-	paddingTop: '16px',
-	borderTop: '1px solid #e0e0e0',
-} );
+export const SftpCard = ( { disabled }: SftpCardProps ) => {
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+
+	const siteId = useSelector( getSelectedSiteId );
+	const siteSlug = useSelector( getSelectedSiteSlug );
+	const currentUserId = useSelector( getCurrentUserId );
+	const siteHasSftpFeature = useSelector( ( state ) =>
+		siteHasFeature( state, siteId, FEATURE_SFTP )
+	);
+	const siteHasSshFeature = useSelector( ( state ) =>
+		siteHasFeature( state, siteId, FEATURE_SSH )
+	);
+	const isSshAccessEnabled =
+		'ssh' === useSelector( ( state ) => getAtomicHostingSshAccess( state, siteId ) );
+	const isLoadingSftpData = useSelector( ( state ) =>
+		getAtomicHostingIsLoadingSftpData( state, siteId )
+	);
+	const { username, password } = useSelector( ( state ) => {
+		let username;
+		let password;
+
+		const users = getAtomicHostingSftpUsers( state, siteId );
+
+		if ( ! disabled && users !== null ) {
+			if ( users.length ) {
+				// Pick first user. Rest of users will be handled in next phases.
+				username = users[ 0 ].username;
+				password = users[ 0 ].password;
+			} else {
+				// No SFTP user created yet.
+				username = null;
+				password = null;
+			}
+		}
+
+		return { username, password };
+	} );
```

Comment:
I copied the old logic exactly because this component relies on a difference between `null` and `undefined` values here (something I learned while fixing the tests). 

Definitely not an ideal solution to begin with, but the scope of this PR is already pretty large, so if we want to fix that, we should do it in another PR (I don't think it warrants fixing right now btw)